test/unit/test_anal_cc.c

```diff
@@ -0,0 +1,69 @@
+#include <r_anal.h>
+
+#include "minunit.h"
+#include "test_sdb.h"
+
+static Sdb *ref_db() {
+	Sdb *db = sdb_new0 ();
+	sdb_set (db, "cc.sectarian.ret", "rax", 0);
+	sdb_set (db, "cc.sectarian.self", "rsi", 0);
+	sdb_set (db, "cc.sectarian.error", "rdi", 0);
+	sdb_set (db, "cc.sectarian.arg1", "rcx", 0);
+	sdb_set (db, "cc.sectarian.arg0", "rdx", 0);
+	sdb_set (db, "cc.sectarian.argn", "stack", 0);
+	sdb_set (db, "sectarian", "cc", 0);
+	return db;
+}
+
+static RAnal *ref_anal() {
+	RAnal *anal = r_anal_new ();
+	r_anal_cc_set (anal, "rax sectarian(rdx, rcx, stack)");
+	r_anal_cc_set_self (anal, "sectarian", "rsi");
+	r_anal_cc_set_error (anal, "sectarian", "rdi");
+	return anal;
+}
+
+bool test_r_anal_cc_set() {
+	RAnal *anal = ref_anal ();
+
+	Sdb *ref = ref_db ();
+	assert_sdb_eq (anal->sdb_cc, ref, "set cc");
+	sdb_free (ref);
+
+	r_anal_free (anal);
+	mu_end;
+}
+
+bool test_r_anal_cc_get() {
+	RAnal *anal = ref_anal ();
+	char *v = r_anal_cc_get (anal, "sectarian");
+	mu_assert_streq (v, "rax sectarian (rdx, rcx, stack, rsi, rdi);", "get cc");
```

Comment:
Just appending rsi and rdi which are actually self and error here seems random to me. Do we really want this? @trufae

Reply:
self an error? for what specific case?

Reply:
wat. Self and error are the registers used for the thisptr and exception stuff. They aren't regular arguments. Also I would expect the result of r_anal_cc_get() to be able to be parsed by r_anal_cc_set() again but this would get you a different result.

Reply:
A bit more context:

The calling convention I saved is:
`rax sectarian (rdx, rcx, stack);` with `rsi` begin used for the thisptr and `rdi` being used for error, so `rsi` and `rdi` are NOT just regular arguments.

But `r_anal_cc_get()` from this gives me:
```rax sectarian (rdx, rcx, stack, rsi, rdi);```
Which would mean that `rsi` and `rdi` would just be regular arguments, which is wrong.

Reply:
Ok got it. We will probably need to use some attributes for handling this insteaf of just the word “stack” like rsi@error and such..

Reply:
Maybe something like this:
```<ret> <self>.<cc name> (<arg0>, <arg1>, <arg2>, <stack/argn>) <error>```

Reply:
updated with the above syntax